File: api-openbis-javascript/src/v3/as/dto/datastore/search/DataStoreSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/common/search/AbstractObjectSearchCriteria",
	"as/dto/datastore/search/DataStoreKindSearchCriteria", "as/dto/datastore/search/DataStoreKind"],
	function(require, stjs, AbstractObjectSearchCriteria, DataStoreKindSearchCriteria, DataStoreKind) {
	var DataStoreSearchCriteria = function() {
		AbstractObjectSearchCriteria.call(this);
	};
	stjs.extend(DataStoreSearchCriteria, AbstractObjectSearchCriteria, [ AbstractObjectSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.datastore.search.DataStoreSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withCode = function() {
			var CodeSearchCriteria = require("as/dto/common/search/CodeSearchCriteria");
			return this.addCriteria(new CodeSearchCriteria());
		};
		prototype.withCodes = function() {
			var CodesSearchCriteria = require("as/dto/common/search/CodesSearchCriteria");
			return this.addCriteria(new CodesSearchCriteria());
		};
		prototype.withKind = function() {
			var DataStoreKindSearchCriteria = require("as/dto/datastore/search/DataStoreKindSearchCriteria");
			this.criteria = this.criteria.filter(criterion => !(criterion instanceof DataStoreKindSearchCriteria));
			return this.addCriteria(new DataStoreKindSearchCriteria());
		};
		prototype.withPermId = function() {
			var PermIdSearchCriteria = require("as/dto/common/search/PermIdSearchCriteria");
			return this.addCriteria(new PermIdSearchCriteria());
		};
		prototype.withSubcriteria = function(subcriteria) {
			return this.addCriteria(subcriteria);
		}
	}, {
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return DataStoreSearchCriteria;
})

File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/datastore/search/DataStoreSearchCriteria.java
/*
 * Copyright ETH 2008 - 2023 Zürich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.datastore.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.CodeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.CodesSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.PermIdSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchOperator;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.datastore.id.IDataStoreId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.datastore.search.DataStoreSearchCriteria")
public class DataStoreSearchCriteria extends AbstractObjectSearchCriteria<IDataStoreId>
{

    private static final long serialVersionUID = 1L;

    public DataStoreSearchCriteria()
    {
    }

    public PermIdSearchCriteria withPermId()
    {
        return with(new PermIdSearchCriteria());
    }

    public CodeSearchCriteria withCode()
    {
        return with(new CodeSearchCriteria());
    }

    public CodesSearchCriteria withCodes()
    {
        return with(new CodesSearchCriteria());
    }

    public DataStoreSearchCriteria withSubcriteria(final DataStoreSearchCriteria subcriteria)
    {
        return with(subcriteria);
    }

    public DataStoreKindSearchCriteria withKind()
    {
        criteria.removeIf(criterion -> criterion instanceof DataStoreKindSearchCriteria);
        return with(new DataStoreKindSearchCriteria());
    }

    public DataStoreSearchCriteria withOrOperator()
    {
        return (DataStoreSearchCriteria) withOperator(SearchOperator.OR);
    }

    public DataStoreSearchCriteria withAndOperator()
    {
        return (DataStoreSearchCriteria) withOperator(SearchOperator.AND);
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("DATA_STORE");
        return builder;
    }

}
